integrated_channels/blackboard/client.py

```diff
@@ -525,37 +552,49 @@ def _get_or_create_integrated_grade_column(self, bb_course_id, grade_column_name
             bb_course_id (str): The Blackboard course ID in which to search for the edX final grade,
             grade column.
         """
-        grade_column_response = self._get(self.generate_gradebook_url(bb_course_id))
-        parsed_response = grade_column_response.json()
-        grade_columns = parsed_response.get('results')
-
+        gradebook_column_url = self.generate_gradebook_url(bb_course_id)
         grade_column_id = None
-        for grade_column in grade_columns:
-            if grade_column.get('externalId') == external_id:
-                grade_column_id = grade_column.get('id')
-        if not grade_column_id:
-            # Potential customization here per-customer, if the need arises.
-            grade_column_data = {
-                "externalId": external_id,
-                "name": grade_column_name,
-                "displayName": grade_column_name,
-                "description": "edX learner's grade.",
-                "externalGrade": False,
-                "score": {
-                    "possible": points_possible
-                },
-                "availability": {
-                    "available": "Yes"
-                },
-                "grading": {
-                    "type": "Manual",
-                    "scoringModel": "Last",
-                    "anonymousGrading": {
-                        "type": "None",
-                    }
-                },
-            }
+        more_pages_present = True
+        current_page_count = 0
+
+        while more_pages_present and current_page_count <= 250:
+            grade_column_response = self._get(gradebook_column_url)
+            parsed_response = grade_column_response.json()
+            grade_columns = parsed_response.get('results')
+
+            for grade_column in grade_columns:
+                if grade_column.get('externalId') == external_id:
+                    grade_column_id = grade_column.get('id')
+                    break
+
+            # Blackboard's pagination is returned within the response json if it exists
+            # Example:
+            # Response = {
+            #     'results': [{
+            #         'id': 1,
+            #         ...
+            #     }, {
+            #         'id', 2,
+            #         ...
+            #     }],
+            #     'paging': {
+            #         'nextPage': '/learn/api/public/v1/courses/<courseID>/gradebook/columns?offset=200'
+            #     }
+            # }
+            if parsed_response.get('paging') and not grade_column_id:
+                gradebook_column_url = '{}{}'.format(
+                    self.enterprise_configuration.blackboard_base_url,
+                    parsed_response.get('paging').get('nextPage'),
+                )
+            else:
+                more_pages_present = False
+
+            current_page_count += 1
```

Comment:
is there a point in incrementing this if we set `more_pages_present = False`
To state another way, why not stick this in the `if` statement above (and I mean the one right above this Else. The other one I understand is within a For loop)

Reply:
TBH you're right, there's no need to increment for both cases. If we have no more pages present then there's no point in incrementing the page count. It doesn't necessarily break anything to have it there, but for the sake of readability I can change.